### PR TITLE
note service v2

### PR DESCRIPTION
**Pull request by zakaria-root**
crud of note element & module
add the endpoint update many notes module & element

### CODE REVIEW

File: note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java [hunk 1]
```diff
@@ -61,11 +63,11 @@ public ResponseEntity<NoteElementWithElementResponse> get(@PathVariable("noteEle
                 .body(foundModuleResponse);
     }
     @GetMapping
-    public  ResponseEntity<NoteElementPagingResponse> getAll(@RequestParam(defaultValue = "0") @Min(0) int page,
-                                                            @RequestParam(defaultValue = "10") @Range(min = 1, max = 10) int size) {
+    public  ResponseEntity<NoteElementPagingResponse> getAllByCodeSession(@RequestParam(defaultValue = "0") @Min(0) int page,
+                                                            @RequestParam(defaultValue = "10") @Range(min = 1, max = 10) int size, @RequestParam String codeSession) {
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java:67]: as said before, there is no need for pagination in this service.

File: note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java [hunk 2]
```diff
@@ -92,6 +94,27 @@ public ResponseEntity<NoteElementResponse> update(
                 .body(updatedModuleResponse);
     }
 
+    @PatchMapping("/bulk")
+    public ResponseEntity<List<NoteElementResponse>> updateMany(
+            @NotEmpty @RequestParam("noteElementId") List<String> noteElementIds,
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java:99]: as said before, you can include ids inside updateRequest body, and the RequestBody must be List<NoteElementUpdateRequest>, we are updating multiple objects, no one.

File: note-service/src/main/java/ma/enset/noteservice/controller/NoteModuleController.java [hunk 3]
```diff
@@ -60,11 +62,11 @@ public ResponseEntity<NoteModuleWithModuleResponse> get(@PathVariable("noteModul
     }
 //
     @GetMapping
-    public ResponseEntity<NoteModulePagingResponse> getAll(@RequestParam(defaultValue = "0") @Min(0) int page,
-                                                            @RequestParam(defaultValue = "10") @Range(min = 1, max = 10) int size) {
+    public ResponseEntity<NoteModulePagingResponse> getAllBycodeSession(@RequestParam(defaultValue = "0") @Min(0) int page,
+                                                            @RequestParam(defaultValue = "10") @Range(min = 1, max = 10) int size, @RequestParam String codeSession) {
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/controller/NoteModuleController.java:66]: Checkout NoteElementController

File: note-service/src/main/java/ma/enset/noteservice/controller/NoteModuleController.java [hunk 4]
```diff
@@ -89,6 +91,27 @@ public ResponseEntity<NoteModuleResponse> update(
                 .body(updatedModuleResponse);
     }
 
+    @PatchMapping("/bulk")
+    public ResponseEntity<List<NoteModuleResponse>> updateMany(
+            @NotEmpty @RequestParam("noteModuleId") List<String> noteModuleIds,
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/controller/NoteModuleController.java:96]: Checkout NoteElementController

**Review by Elbouchouki — CHANGES_REQUESTED** [review]
When refactored, there is no need to recreate a new pull request.

File: note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java (hunk 2 above)
Comment (zakaria-root) [reply note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java:99]: I think it's better to go with request param, and I'm updating a list of object, not one!

File: note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java (hunk 1 above)
Comment (zakaria-root) [reply note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java:67]: Sorry I forgot it

File: note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java (hunk 2 above)
Comment (Elbouchouki) [reply note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java:103]: you are passing list of ids and a single object. you update all the objects associated with those ids with the updateRequest you send in body. that isn't right. you must provide list of updateRequest and for each id you update its object with one of the updateRequests . (checkout bulk update in other services)

File: note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java [hunk 5]
```diff
@@ -27,22 +29,64 @@ public class NoteElementController {
...
+    private final NoteModuleService noteModuleService;
 
     @PostMapping
     public ResponseEntity<NoteElementResponse> save(@Valid @RequestBody NoteElementCreationRequest noteElementCreationRequest) {
-        NoteElement note = noteElementMapper.toNoteElement(noteElementCreationRequest);
-        NoteElementResponse noteElementResponse = noteElementMapper.toNoteElementResponse(noteElementService.save(note));
+        NoteElement noteElement = noteElementMapper.toNoteElement(noteElementCreationRequest);
+        NoteElementResponse noteElementResponse = noteElementMapper.toNoteElementResponse(noteElementService.save(noteElement));
+
+        ResponseEntity<ElementResponse> elementResponse =  elementServiceFeignClient.getElementByCode(noteElement.getCodeElement());
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java:39]: this logic should be moved to service

File: note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java [hunk 6]
```diff
@@ -27,22 +29,64 @@ public class NoteElementController {
...
     @PostMapping("/bulk")
     public ResponseEntity<List<NoteElementResponse>> saveAll(@RequestBody List<@Valid NoteElementCreationRequest> noteElementCreationRequestList) {
+//
+
         List<NoteElement> noteElementList = noteElementMapper.toNoteElementList(noteElementCreationRequestList);
         List<NoteElementResponse> noteElementResponseList = noteElementMapper.toNoteElementResponseList(noteElementService.saveAll(noteElementList));
 
+        List<NoteModule> noteModuleList = new ArrayList<>();
+        noteElementList.forEach( noteElement -> {
+            ResponseEntity<ElementResponse> elementResponse =  elementServiceFeignClient.getElementByCode(noteElement.getCodeElement());
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java:69]: this logic should be moved to service

File: note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java [hunk 7]
```diff
@@ -85,6 +130,21 @@ public ResponseEntity<NoteElementResponse> update(
         noteElementMapper.updateNoteElementFromDTO(noteElementUpdateRequest, noteElement);
         NoteElementResponse updatedModuleResponse = noteElementMapper.toNoteElementResponse(noteElementService.update(noteElement));
 
+        ResponseEntity<ElementResponse> elementResponse =  elementServiceFeignClient.getElementByCode(noteElement.getCodeElement());
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java:133]: this logic should be moved to service

File: note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java [hunk 8]
```diff
@@ -98,6 +158,30 @@ public ResponseEntity<List<NoteElementResponse>> updateAll(
         List<NoteElement> noteElementList = noteElementService.findAllByNoteElementId(noteElementMapper.toNoteElementIdList(noteElementUpdateRequestList));
         noteElementMapper.updateNoteElementsFromDTO(noteElementUpdateRequestList, noteElementList);
         List<NoteElement> updatedNoteElementList = noteElementService.updateAll(noteElementList);
+
+        List<NoteModule> noteModuleList = new ArrayList<>();
+        noteElementList.forEach(noteElement -> {
+            ResponseEntity<ElementResponse> elementResponse =  elementServiceFeignClient.getElementByCode(noteElement.getCodeElement());
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java:164]: this logic should be moved to service

**Review by Elbouchouki — CHANGES_REQUESTED** [review]

**Comment by Elbouchouki** [comment]
closed in favor of [Improve note service](https://github.com/projet-d-innovation/delib-backend/issues/75)

**Pull request closed without merging**